Add Event Analytics visualizations in Dashboards feature

vagimeli opened the pull request:
### Description
Add Event Analytics visualizations to existing documentation. 

### Issues Resolved
#3549 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

vagimeli commented:
@joshuali925 Will you review this first-pass draft on the event analytics dashboards view feature? If possible, I'd appreaciate your feedback by Tuesday, 4/ 18. Please check for technical accuracy of the information and let me know if there are any information gaps. If we need to sync again, please let me know. Thanks for your support!

joshuali925 reviewed (commented):
thanks for the changes.

One thing I forgot to mention in detail is the limitation on filtering. PPL visualizations currently do not support DQL/DSL and they do not use index patterns. This means

1. PPL visualizations will only use filters created using the dropdown UI. If user has DQL query or DSL filters in a dashboard, PPL visualizations will not use them
2. OSD dashboard filter dropdown will only show fields from the index patterns used by other visualizations in the same dashboard (or from default index pattern if PPL visualization are the only visualizations in the dashboard).

Should we mention this to user in docs or should I create a tracking issue and put it in observability readme?

Some related information here https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3849

kolchfa-aws commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,61 @@ For example, the following PPL query returns a count of how many host addresses
-## View logs
+## Adding event analytics visualizations to dashboards
+
+Presenting your visualizations on a dashboard, instead of the event analytics page, makes it easier for users with varying levels of technical knowledge to understand and interpret the data at a glance. To create a new visualization and add it to a dashboard, follow these steps:
```
I would remove "for users with varying levels of technical knowledge"

kolchfa-aws commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,61 @@ For example, the following PPL query returns a count of how many host addresses
+Presenting your visualizations on a dashboard, instead of the event analytics page, makes it easier for users with varying levels of technical knowledge to understand and interpret the data at a glance. To create a new visualization and add it to a dashboard, follow these steps:
+
+1. On the main menu, select **Observability** > **Event analytics**.
+2. On the Event analytics window, enter the index source in the PPL query field, for example, `source = opensearch_dashboards_sample_data_logs | stats max(bytes), avg(bytes) by host`.
```
*in* the event analytics window? Should "PPL query" be bold if it's a UI element? Should we specify that you have to enter it in the PPL language or is it obvious because of the field name?

kolchfa-aws commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,61 @@ For example, the following PPL query returns a count of how many host addresses
+5. Select **Save** and enter a name for the visualization. To add the visualization to an existing dashboard, choose from the **Custom operational dashboards/application** dropdown menu. 
+6. Select **Save**.
+
+You've created a new visualization that can be added to an existing or new dashboard. To work with the dashboard view, follow these steps:  
```
Should we be more specific here and say "To add a PPL query to a dashboard" instead of "to work with the dashboard view"?

kolchfa-aws commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,61 @@ For example, the following PPL query returns a count of how many host addresses
+
+PPL visualizations currently do not support [Dashboards Query Language (DQL)]({{site.url}}{{site.baseurl}}/dashboards/discover/dql/) or [query domain-specific language (DSL)]({{site.url}}{{site.baseurl}}/query-dsl/) and they do not use index patterns. Note the following limitations: 
+
+- PPL visualizations will only use filters created using the dropdown interface. If you have DQL query or DSL filters in a dashboard, PPL visualizations will not use them.
```
Should this be in present tense?

kolchfa-aws commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,61 @@ For example, the following PPL query returns a count of how many host addresses
+PPL visualizations currently do not support [Dashboards Query Language (DQL)]({{site.url}}{{site.baseurl}}/dashboards/discover/dql/) or [query domain-specific language (DSL)]({{site.url}}{{site.baseurl}}/query-dsl/) and they do not use index patterns. Note the following limitations: 
+
+- PPL visualizations will only use filters created using the dropdown interface. If you have DQL query or DSL filters in a dashboard, PPL visualizations will not use them.
+- The **Dashboard** filter dropdown will only show fields from the index patterns used by other visualizations in the same dashboard (or from the default index pattern if PPL visualizations are the only visualization types in the dashboard).
```
Same: present tense?

joshuali925 replied:
I realized this might not be accurate as there can be non-PPL visualizations but also not using index patterns, like visbuilder in 2.6 which had the same limitation

```suggestion
- The **Dashboard** filter dropdown interface will only show fields from the default index pattern or index patterns used by other visualizations in the same dashboard.
```

kolchfa-aws reviewed (approved):
LGTM. Left a couple of open questions.

natebower commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,65 @@ For example, the following PPL query returns a count of how many host addresses
+1. On the main menu, select **Observability** > **Event analytics**.
+2. In the Event analytics window, enter the index source in the **PPL query** field, for example, `source = opensearch_dashboards_sample_data_logs | stats max(bytes), avg(bytes) by host`. You must enter the query using PPL syntax. 
+3. Set the time filter, for example **Week to date**, then select **Refresh**.
+4. Choose the visualization type, for example **Time series**, from the right sidebar dropdown menu.  
```
```suggestion
4. Choose the visualization type, for example, **Time series**, from the right sidebar dropdown menu.  
```

natebower commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,65 @@ For example, the following PPL query returns a count of how many host addresses
+5. Select **Save** and enter a name for the visualization. To add the visualization to an existing dashboard, choose from the **Custom operational dashboards/application** dropdown menu. 
+6. Select **Save**.
+
+You've created a new visualization that can be added to an existing or new dashboard. To add a PPL query to a dashboard, follow these steps:  
```
```suggestion
You've created a new visualization that can be added to a new or existing dashboard. To add a PPL query to a dashboard, follow these steps:  
```

natebower commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,65 @@ For example, the following PPL query returns a count of how many host addresses
+
+You've created a new visualization that can be added to an existing or new dashboard. To add a PPL query to a dashboard, follow these steps:  
+
+1. On the main menu, select **Dashboard**.
```
```suggestion
1. Select **Dashboard** from the main menu.
```

natebower commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,65 @@ For example, the following PPL query returns a count of how many host addresses
+You've created a new visualization that can be added to an existing or new dashboard. To add a PPL query to a dashboard, follow these steps:  
+
+1. On the main menu, select **Dashboard**.
+2. On the **Dashboards** window, select **Create dashboard**.
```
```suggestion
2. In the **Dashboards** window, select **Create dashboard**.
```

natebower commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,65 @@ For example, the following PPL query returns a count of how many host addresses
+
+1. On the main menu, select **Dashboard**.
+2. On the **Dashboards** window, select **Create dashboard**.
+3. On the Dashboards window, select **Add existing** and then choose the visualization.
```
```suggestion
3. In the **Dashboards** window, select **Add existing** and then choose the visualization.
```

natebower commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,65 @@ For example, the following PPL query returns a count of how many host addresses
+1. On the main menu, select **Dashboard**.
+2. On the **Dashboards** window, select **Create dashboard**.
+3. On the Dashboards window, select **Add existing** and then choose the visualization.
+4. To add more visualizations to the dashboard, select **Create new** and then select **PPL** on the New Visualization window. You'll return to the event analytics page and follow steps 1-6 in the preceding instructions.
```
```suggestion
4. To add more visualizations to the dashboard, select **Create new** and then select **PPL** in the **New Visualization** window. You'll return to the event analytics page and follow steps 1--6 in the preceding instructions.
```

natebower commented on _observing-your-data/event-analytics.md:
```diff
@@ -24,33 +24,65 @@ For example, the following PPL query returns a count of how many host addresses
+
+### Limitations of event analytics visualizations
+
+Event analytics visualizations currently do not support [Dashboards Query Language (DQL)]({{site.url}}{{site.baseurl}}/dashboards/discover/dql/) or [query domain-specific language (DSL)]({{site.url}}{{site.baseurl}}/query-dsl/) and they do not use index patterns. Note the following limitations: 
```
```suggestion
Event analytics visualizations currently do not support [Dashboards Query Language (DQL)]({{site.url}}{{site.baseurl}}/dashboards/discover/dql/) or [query domain-specific language (DSL)]({{site.url}}{{site.baseurl}}/query-dsl/), and they do not use index patterns. Note the following limitations: 
```

vagimeli commented:
> @vagimeli Please see my comments and changes and let me know if you have any questions. Thanks!

Accepted editorial review changes and addressed comments.
